### PR TITLE
[FIX] unlink bi_sql_view without cron

### PR DESCRIPTION
**Pull request by PierrickBrun**
@legalsylvain trivial fix PR

### CODE REVIEW

File: bi_sql_editor/models/bi_sql_view.py [hunk 1]
```diff
@@ -263,7 +263,8 @@ def unlink(self):
                     "If you want to delete them, first set them to draft."
                 )
             )
-        self.cron_id.unlink()
+        if self.cron_id:
+            self.cron_id.unlink()
```
Comment (legalsylvain) [inline bi_sql_editor/models/bi_sql_view.py:267]: Maybe replace two lines by

self.mapped("cron_id").unlink()

Otherwise, LGTM

Comment (PierrickBrun) [reply bi_sql_editor/models/bi_sql_view.py:267]: I've just tested, it does not work either.

I think the real problem is that unlink() crashes if called from an empty recordset

Comment (PierrickBrun) [reply bi_sql_editor/models/bi_sql_view.py:267]: probably only on ir_cron, that said: https://github.com/odoo/odoo/blob/16.0/odoo/addons/base/models/ir_cron.py#L401

Comment (PierrickBrun) [reply bi_sql_editor/models/bi_sql_view.py:267]: I've created this PR: https://github.com/odoo/odoo/pull/125232

Comment (legalsylvain) [reply bi_sql_editor/models/bi_sql_view.py:267]: ah yes. makes sense.

Comment (leemannd) [reply bi_sql_editor/models/bi_sql_view.py:267]: Hello, open question, why not setting the field as `ondelete=cascade` ?

**Pull request merged**